Orchestration mode

ArcturusZhang opened the pull request.

fitzgeraldsteele commented on website/docs/r/virtual_machine_scale_set_vm_mode.html.markdown:
```diff
@@ -0,0 +1,130 @@
+  name                = "example-VMSS"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
```
can also set singleplacementgroup true/false

This will be in private preview for Walmart until Sept.\
Current default should be true.

Can just make this a required parameter and customer can set

ArcturusZhang replied:
DOne

fitzgeraldsteele commented on website/docs/r/virtual_machine_scale_set_vm_mode.html.markdown:
```diff
@@ -0,0 +1,130 @@
+
+* `id` - The ID of the Virtual Machine Scale Set VM Mode.
+
+* `unique_id` - The Unique ID for this Virtual Machine Scale Set VM Mode.
```
I'm not sure how id and unique_id are different?

ArcturusZhang replied:
ID is the full resource ID of this VMSS, like `subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Compute/virtualMachineScaleSets/scaleset1`
And an unique ID is an ID of this VMSS (but I do not quite know where this will be used really), should be in UUID format. And a VM has the same thing called `VMID`

sirlatrom committed:
Add health_check_path to site_config

Signed-off-by: Sune Keller <absukl@almbrand.dk>

sirlatrom committed:
Add acceptance test for health_check_path

Signed-off-by: Sune Keller <absukl@almbrand.dk>

sirlatrom committed:
Document health_check_path on azurerm_app_service

Signed-off-by: Sune Keller <absukl@almbrand.dk>

Commit:
fix comments

Commit:
fix comments 2nd round

Commit:
add pre_warmed_instance_count

Commit:
Update code

r0bnet committed:
start server when it was stopped before updating it

r0bnet committed:
fix lintrest problem

Commit:
Update doc for property default_action of eventhub namespace

arestarh committed:
Changed: notes regarding configuring 'TLS termination with Key Vault certificates' for App Gateway were added.

jackofallops committed:
Merge pull request #6624 from neil-yechenwei/fix_nic

jackofallops committed:
Update for #6624

tombuildsstuff committed:
Merge pull request #6785 from arestarh/azurerm_app_gw_ssl_cert_key_vault_doc

Changed: notes regarding configuring 'TLS termination with Key Vault …

r0bnet committed:
Merge branch 'master' into analysis-services-server-update-when-paused

r0bnet committed:
fix PR recommendations

jackofallops committed:
Merge pull request #6774 from r0bnet/analysis-services-server-ipv4-rule-name-case-fix

jackofallops committed:
Update for #6774

tombuildsstuff committed:
Merge pull request #6747 from Avinode/recovery-network-mapping-fix-for-issue-6026

Workaround for strange response code in recovery services network mappings API. Fixes issue #6026

tombuildsstuff committed:
updating to include #6747

jackofallops committed:
Merge pull request #6435 from aristosvo/functionslot

jackofallops committed:
Update for #6435

r0bnet committed:
add test for updating a paused analysis services server

tracypholmes committed:
Updates `apimanagement`

tracypholmes committed:
Updates `applicationinsights`

tracypholmes committed:
Updates `appplatform`

tracypholmes committed:
Updates `authorization`

tracypholmes committed:
Updates `automation`

tracypholmes committed:
Updates `batch`

tracypholmes committed:
Updates `bot`

tracypholmes committed:
Updates `cdn`

tracypholmes committed:
Updates `cognitive`

tracypholmes committed:
Updates `compute`

tracypholmes committed:
Updates `containers`

tracypholmes committed:
Updates `cosmos`

tracypholmes committed:
Updates `costmanagement`

tracypholmes committed:
Updates `customproviders`

tracypholmes committed:
Updates `databasemigration`

tracypholmes committed:
Updates `databricks`

tracypholmes committed:
updates `datafactory`

tracypholmes committed:
Updates `datalake`

tracypholmes committed:
Updates `datashare`

tracypholmes committed:
Updates `devspace`

tracypholmes committed:
Updates `devtestlabs`

tracypholmes committed:
Updates `dns`

tracypholmes committed:
fmts & builds

nickmhankins committed:
azurerm_eventhub_namespace_authorization_rule  - lock so multiple resource updates won't clash (#6701)

Fixes #4893

katbyte committed:
update CHANGELOG.md to include #6701

katbyte committed:
Merge pull request #6797 from terraform-providers/tph/file-rename

magodo committed:
`azurerm_monitor_diagnostic_setting` - `log_analytics_destination_type` supports `AzureDiagnostics` #6769

mbfrahry committed:
Update for #6769

tracypholmes committed:
updates `eventgrid`

tracypholmes committed:
updates `eventhub`

tracypholmes committed:
updates `frontdoor`

tracypholmes committed:
updates `hdinsight`

tracypholmes committed:
updates `healthcare`

tracypholmes committed:
updates `iotcentral`

tracypholmes committed:
updates `iothub`

tracypholmes committed:
Updates `keyvault`

tracypholmes committed:
updates `kusto`

tracypholmes committed:
updates `loganalytics`

tracypholmes committed:
updates `logic`

tracypholmes committed:
updates `machinelearning`

tracypholmes committed:
updates `maintenance`

tracypholmes committed:
updates `managementgroup`

tracypholmes committed:
updates `maps`

tracypholmes committed:
updates `mariadb`

tracypholmes committed:
updates `media`

tracypholmes committed:
updates `mixedreality`

tracypholmes committed:
updates `monitor`

tracypholmes committed:
updates `msi`

tracypholmes committed:
updates `mssql`

tracypholmes committed:
updates `mysql`

tracypholmes committed:
updates `netapp`

tracypholmes committed:
updates `notificationhub`

tracypholmes committed:
updates `network`

tracypholmes committed:
updates `policy`

tracypholmes committed:
updates `portal`

tracypholmes committed:
updates `postgres`

tracypholmes committed:
updates `powerbi`

tracypholmes committed:
updates `privatedns`

katbyte committed:
Merge pull request #6798 from terraform-providers/tph/moar-file-rename

katbyte committed:
Merge pull request #6799 from terraform-providers/tph/extra-moar-file-rename

katbyte committed:
Merge pull request #6800 from terraform-providers/tph/some-moar-file-rename

aqche committed:
address review

aqche committed:
formatting

aqche committed:
formatting

aqche committed:
formatting

magodo committed:
Update website/docs/r/virtual_hub_connection.html.markdown

Co-authored-by: Matthew Frahry <mbfrahry@gmail.com>

Commit:
Update doc for property sku of servicebus namespace

r0bnet committed:
Merge branch 'master' into analysis-services-server-update-when-paused

r0bnet committed:
add note that analysis services server is started and stopped during update

tombuildsstuff committed:
Merge pull request #6766 from magodo/vhub_doc

Add documentation about virtual hub connection field `vitual_network_to_hub_gateways_traffic_allowed`

tombuildsstuff committed:
Merge pull request #6806 from neil-yechenwei/fix_eventhub_namespace

Update doc for property default_action of eventhub namespace

tombuildsstuff committed:
Merge pull request #6804 from neil-yechenwei/fix_doc_eventhubns

Update doc for property sku of servicebus namespace

tombuildsstuff committed:
Merge pull request #6786 from r0bnet/analysis-services-server-update-when-paused

[Enhancement] Allow updating Analysis Services Server when paused

tombuildsstuff committed:
updating to include #6786

jackofallops committed:
fix broken test param for remoteDebugging

jackofallops committed:
fixes some broken tests

jackofallops committed:
update validation for new value in remoteDebuggingVersion

sirlatrom committed:
Note that App Service Health Check is in Preview

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

jackofallops committed:
added todo

jackofallops committed:
Merge pull request #6812 from terraform-providers/t/web-test-fixes-20200507

jackofallops committed:
Merge pull request #6661 from sirlatrom/fix-6429

jackofallops committed:
Update for #6661

jackofallops committed:
Merge pull request #6705 from aqche/app_service_ip_restriction_name_and_priority

jackofallops committed:
Update for #6705

magodo committed:
New resource: `azurerm_sentinel_alert_rule_scheduled` #6650

mbfrahry committed:
Update for #6650

tombuildsstuff committed:
builder: raising an error if environment is unintentionally set to "AZURESTACKCLOUD"

Users should instead use the separate "azurestack" provider which offers integration
with Azure Stack via the Azure Stack Profiles

Commit:
support alias connection strings for eventhub resources (#6708)

katbyte committed:
update CHANGELOG.md to include #6708

tombuildsstuff committed:
Merge pull request #6817 from terraform-providers/b/azure-stack

builder: raising an error for unsupported environments

tombuildsstuff committed:
updating to include #6817 / formatting

njuCZ committed:
update doc for "azurerm_redis_cache" (#6807)

ArcturusZhang committed:
update `azurerm_policy_assignment` - allow scopes of policy set definition without `subscription/<id>` (#6792)

Fixes #6671

katbyte committed:
update CHANGELOG.md to include #6792

njuCZ committed:
`azurerm_automation_runbook` - add support for create runbook as draft and then publish  (#6813)

fixes #6588
fixes #4403
fixes #3086

katbyte committed:
update CHANGELOG.md to include #6813

cemckenzie committed:
[BUG] azurerm_redis_cache backup requires connection string (#6819)

katbyte committed:
update CHANGELOG.md to include #6819

Commit:
v2.9.0

Commit:
Cleanup after v2.9.0 release

ArcturusZhang committed:
Merge remote-tracking branch 'origin/master' into Orchestration-mode

ArcturusZhang committed:
Rename virtual_machine_scale_set_orchestrator_vm to orchestrated_virtual_machine_scale_set

lrxtom2 committed:
Update express_route_circuit_peering.html.markdown (#6824)

yupwei68 committed:
`azurerm_mssql_server`  - Add support for `azuread_administrator` (#6822)

mbfrahry committed:
Update for #6822

tracypholmes committed:
docs: added note to `infrastructure_encryption_enabled` property (#6831)

magodo committed:
`azurerm_automation_account` DS set its own ID

Previously, it uses the automation account agent registration info's ID
as its ID, which cause other resource which refers to that ID will fail
to parse the resource ID (as it doesn't apply to resource ID format).

Nevertheless, we shall change to use the ID of resource
`azurerm_automation_account` as the DS's ID.

dragz committed:
Fix load-balancer example with new syntax for v2.0 (#6836)

robselway committed:
New data source azurerm_eventhub

robselway committed:
Removed space from Event Hub in docs

magodo committed:
modify regexp: escape dot

robselway committed:
Fixed tflint errors

robselway committed:
Added azurerm_eventhub data source to website navigation

robselway committed:
tf format in website docs

tombuildsstuff committed:
Merge pull request #6848 from magodo/automation_account_ds_id

`azurerm_automation_account` DS set its own ID

tombuildsstuff committed:
updating to include #6848

robselway committed:
Changed partition_ids from TypeSet to TypeList

ArcturusZhang committed:
`data.azurerm_virtual_machine` - export `identity` attribute (#6826)

mbfrahry committed:
Update changelog for #6826

tombuildsstuff committed:
Merge pull request #6841 from robselway/feature/eventhub_data_source

New data source azurerm_eventhub

tombuildsstuff committed:
updating to include #6841

Commit:
Updates doc for aks node pool (#6825)

jrauschenbusch committed:
Upgrade EventGrid SDK to 2020-04-01-preview (#6837)

Upgrade EventGrid SDK to 2020-04-01-preview from 2018-09-15-preview to address @katbyte comments in PR #5948.

katbyte committed:
update CHANGELOG.md to include #6837

mbfrahry committed:
Fix crash

ml4 committed:
Fix typo in terraform state mv command (#6856)

mbfrahry committed:
fmt

hbuckle committed:
`azurerm_application_insights` - `retention_in_days` defaults to 90 #6851

mbfrahry committed:
Update changelog for #6851

ArcturusZhang committed:
`azurerm_windows_virtual_machine` - Allow update of OS disk encryption settings (#6846)

Fixes #6737

This applies the changes in PR #6230 to windows virtual machine which only fixes the problem in linux virtual machine despite its title says linux|windows.

katbyte committed:
update CHANGELOG.md to include #6846

magodo committed:
`SchemaResourceGroupNameForDataSource` Add `ValidateFunc`

Commit:
Update doc for traffic manager endpoint

tombuildsstuff committed:
Merge pull request #6864 from magodo/ds_resgroupname_validate

`SchemaResourceGroupNameForDataSource` Add `ValidateFunc`

tombuildsstuff committed:
updating to include #6864

tombuildsstuff committed:
Merge pull request #6862 from terraform-providers/c-api-management

`azurerm_api_management` - fix crash with `policy`

tombuildsstuff committed:
updating to include #6862

tombuildsstuff committed:
ordering

magodo committed:
Update per review

[Link](https://github.com/terraform-providers/terraform-provider-azurerm/pull/6821#pullrequestreview-409203935)

sdebruyn committed:
add type PipelineReference in trigger res

When the type is omitted, the trigger cannot be activated. This fixes #6869

ArcturusZhang committed:
Fix #6830

ArcturusZhang committed:
Fix fmt flag

Commit:
update

Commit:
align on versin 41.2 of SDK

tombuildsstuff committed:
Merge pull request #6871 from datarootsio/bugfix/fix-6869-type-pipelinereference

Add type PipelineReference in trigger resource

tombuildsstuff committed:
updating to include #6871

tombuildsstuff committed:
Merge pull request #6870 from neil-yechenwei/trafficmanagerfix

Update doc for traffic manager endpoint

rudolphjacksonm committed:
Fix case on AuthorizationRules to match Azure API

The resource ID for authorization rule specifies the rule name parameter `authorizationRules` with the first letter capitalized: `AuthorizationRules`.

When trying to import an authorization rule using the resource ID shown in the docs, you'll get the following error:
```
Error: Error making Read request on Azure EventHub Authorization Rule : eventhub.NamespacesClient#GetAuthorizationRule: Invalid input: autorest/validation: validation failed: parameter=authorizationRuleName constraint=MinLength value="" details: value length must be greater than or equal to 1
```

tombuildsstuff committed:
Merge pull request #6875 from mguirao/update-iothub-sdk

dependencies: updating iothub to `2019-03-22-preview`

tombuildsstuff committed:
updating to include #6875

tombuildsstuff committed:
Merge pull request #6878 from rudolphjacksonm/patch-2

Fix case on AuthorizationRules to match Azure API

aristosvo committed:
Bug: azurerm_kubernetes_cluster errors out with InvalidLoadbalancerProfile on changes (#6534)

Fixes #6525

katbyte committed:
update CHANGELOG.md to include #6534

magodo committed:
`azurerm_log_analytics_workspace` - Support `rentention_in_days` for Free Tier (#6844)

mbfrahry committed:
Update changelog for #6844

Commit:
Update doc for network profile of aks (#6810)

fixes #5298

It only supports below scenarios:
network_plugin = azure, network_policy = azure/calico
network_plugin = kubenet, network_policy = calico

yupwei68 committed:
azurerm_api_management_named_value - fix the non empty plan when `secret` is true (#6834)

Fix #6688

katbyte committed:
update CHANGELOG.md to include #6834

Commit:
Update import command in eventhub ns doc

Commit:
update doc for `azurerm_role_assignment`

Commit:
update

jackofallops committed:
Merge pull request #6821 from magodo/ase_resgroup

jackofallops committed:
Update for #6821

tombuildsstuff committed:
Merge pull request #6885 from njuCZ/issue_4430

update doc for `azurerm_role_assignment`

tombuildsstuff committed:
Merge pull request #6886 from neil-yechenwei/fixeventhubnsdoc

Update import command in eventhub ns doc

tombuildsstuff committed:
tool: adding a check for gradually deprecated functions

Go's linter immediately flags usages of functions which are marked as
Deprecated - which makes sense when there's 10 usages, but less so when
there's hundreds of usages and you want to remove things gradually.

This PR introduces a check for functions which will be deprecated in the
future, since they're no longer required - namely the feature flag
"RequireResourcesToBeImported" which is now enabled by default.

Over time we'll work through the existing codebase and remove usages of
this function - but we intentionally don't want to do this in a single
sweep as to not break open PR's (since this defaulted to "enabled" there's
not much harm in this code kicking around for a while)

tombuildsstuff committed:
Merge pull request #6889 from terraform-providers/f/gradually-deprecated-functions

tool: adding a check for gradually deprecated functions

floushee committed:
size property must not be defined if source_uri is used (#6897)

mbfrahry committed:
`data.azurerm_security_group` - fixing crash where id is nil #6910

mbfrahry committed:
Update changelog for #6910

sirlatrom committed:
azurerm_application_gateway  - support WAF Policies (#6105)

Fixes #4667.

katbyte committed:
update CHANGELOG.md to include #6105

sirlatrom committed:
Support the Developer Portal type of hostname_configuration in azurerm_api_management (#6724)

Fixes #6714.

katbyte committed:
update CHANGELOG.md to include #6724

Commit:
azurerm_express_route_circuit - de-provision and re-provision circuit when changing bandwidth reduction (#3983) (#6601)

Reduction in bandwidth forces new resource, utilizing CustomizeDiff

Fixes #3983

katbyte committed:
update CHANGELOG.md to include #6601

WodansSon committed:
fixes #5864 and #6763

* Adding private endpoint example HCL

* Update examples/private_endpoint/postgreSql/README.md

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

* Update examples/private_endpoint/cosmosDb/README.md

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

* Requested changes per PR review

* Update location descriptions in README.md

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

sirlatrom committed:
azurerm_api_management - Only set `hostname_configuration` properties if not empty (#6850)

When the fields are set to empty strings instead of null, the error message in #6621 is given. When set to null when empty strings, the update goes through.

Fixes #6621.

katbyte committed:
update CHANGELOG.md to include #6850

sirlatrom committed:
api_managment - support user assigned managed identities (#6783)

Fixes #6782.

katbyte committed:
update CHANGELOG.md to include #6783

Commit:
Update doc for validation of aks (#6884)

mbfrahry committed:
`azurerm_monitor_diagnostic_setting` - Fix crash around `retention_policy` (#6911)

Fixes #6908

katbyte committed:
update CHANGELOG.md to include #6911

jrauschenbusch committed:
New Resource: `azurerm_eventgrid_domain_topic` (#6859)

Adds new resource azurerm_eventgrid_domain_topic

Fixes #4097

katbyte committed:
update CHANGELOG.md to include #6859

manicminer committed:
mariadb_server: New attributes, flatten `storage_profile` block (#6865)

katbyte committed:
update CHANGELOG.md to include #6865

manicminer committed:
mysql_server: New attributes, flatten `storage_profile` block (#6833)

create_mode, creation_source_server_id,
infrastructure_encryption_enabled,
ssl_minimal_tls_version_enforced

Support replicas and point-in-time restores

Use booleans for toggle properties, renaming

auto_grow -> auto_grow_enabled
geo_redundant_backup -> geo_redundant_backup_enabled
ssl_enforcement -> ssl_enforcement_enabled
Remove redundant storage_profile block and move all properties
within to top level

Fixes: #4756
Related: #6459

katbyte committed:
update CHANGELOG.md to include #6833

katbyte committed:
add documentation for the source_image_reference block

WodansSon committed:
Fixes #6741

* Add support for frontdoor_id computed attribute

* Updated name of new attribute

WodansSon committed:
updating to include #6916

tombuildsstuff committed:
Merge pull request #6918 from terraform-providers/kt/document-the-source

add documentation for the source_image_reference block

HappyTobi committed:
fixes #5758

* Add resource migration for Storage Account

* Update docs to show new behavior

* Update config name to indicate it's a prepare step

Co-authored-by: Jeffrey Cline <20408400+WodansSon@users.noreply.github.com>

WodansSon committed:
updating to include #6580

MrDanao committed:
Fix azurerm_subnet_nat_gateway_association documentation title

jackofallops committed:
Added poller to check for consistency on role assignment creation

jackofallops committed:
Update azurerm/internal/services/authorization/role_assignment_resource.go

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

jackofallops committed:
Merge pull request #6925 from terraform-providers/e/role-assignment-consistency

jackofallops committed:
Update for #6925

tombuildsstuff committed:
Merge pull request #6923 from MrDanao/subnet-nat-gateway-assoc-doc-title

Fix azurerm_subnet_nat_gateway_association documentation title

tombuildsstuff committed:
Merge pull request #6917 from ArcturusZhang/fix-vmss-overprovisioned

r/`azurerm_linux|windows_virtual_machine_scale_set` - updating do_not_run_extensions_on_overprovisioned_machines

tombuildsstuff committed:
updating to include #6917

katbyte committed:
update CHANGELOG.md to include #6723

sirlatrom committed:
azurerm_api_management - support key_vault_id's without a version (#6723)

Fixes #4408.

katbyte committed:
azurerm_postgres_server - support for threat_detection_policy (#6721)

katbyte committed:
update CHANGEloG.md to include #6721

lrxtom2 committed:
Fix duplicate provider configuration in storage tests (#6748)

yupwei68 committed:
`azurerm_api_management_diagnostic` - support required property `api_management_logger_id` (#6682)

fix #6619

This will be a breaking change for it adds a required field "logger_id"

=== RUN TestAccAzureRMApiManagementDiagnostic_basic
=== PAUSE TestAccAzureRMApiManagementDiagnostic_basic
=== CONT TestAccAzureRMApiManagementDiagnostic_basic
--- PASS: TestAccAzureRMApiManagementDiagnostic_basic (2498.17s)
=== RUN TestAccAzureRMApiManagementDiagnostic_update
=== PAUSE TestAccAzureRMApiManagementDiagnostic_update
=== CONT TestAccAzureRMApiManagementDiagnostic_update
--- PASS: TestAccAzureRMApiManagementDiagnostic_update (2566.13s)
=== RUN TestAccAzureRMApiManagementDiagnostic_requiresImport
=== PAUSE TestAccAzureRMApiManagementDiagnostic_requiresImport
=== CONT TestAccAzureRMApiManagementDiagnostic_requiresImport
--- PASS: TestAccAzureRMApiManagementDiagnostic_requiresImport (2330.79s)
PASS

(also fixes #6104)

katbyte committed:
update CHANGELOG.md to include #6682

Commit:
v2.10.0

Commit:
Cleanup after v2.10.0 release

ArcturusZhang committed:
Resolve comments

ArcturusZhang committed:
Fix the bad link

WodansSon committed:
`azurerm_postgresql_server` - Update documentation per request from service team (#6963)

philippeckel committed:
Fix typo in resource "azurerm_application_insights": application_type "Other" -> "other" (#6946)

lrxtom2 committed:
Update doc for cosmosdb account GlobalDocumentDB connection string (#6965)

marcin-dudek committed:
Update eventgrid_event_subscription.html.markdown (#6953)

lrxtom2 committed:
Update doc for custom_domain in storage_account (#6947)

katbyte committed:
provider: upgrade azure-sdk-for-go to v42.1.0 & azuread to v0.9.0 (#6725)

katbyte committed:
update CHANGELOG.md to include #6725

katbyte committed:
upgrade network to 2020-03-01 (#6727)

katbyte committed:
update CHANGEOG.md to include #6727

catay committed:
mariadb_server: remove deprecated interpolation syntax in docs (#6970)

ArcturusZhang committed:
Merge remote-tracking branch 'origin/master' into Orchestration-mode

ArcturusZhang committed:
Fix bad test cases
